### DISCUSSION TITLE
How to untag all tagged files in one remote command?

### DISCUSSION THREAD

**Discussion by jpggithub** [gokcehan/lf#1352] [General]
Hi

I tags all files that I want to attach to a mail (Emacs/mu4e/message-mode).
I already can attach the tagged files by reading the `~/.local/share/lf/tags` file.

Now I want to untag all the tagged files, for instance to not attach them again if I want to write another mail.
My first idea is to use something like `lf -remote 'send untag'` but there is no command like this.
How can I do that?

**Comment by joelim-work** [comment]
Tags are cleared using the `tag-toggle` command. This action applies to the list of selected files if there are any (e.g. via the `toggle` command by pressing <kbd>space</kbd>), otherwise it applies to the current file located at the cursor. Which means you should be able to programatically select some files and then use `tag-toggle`:

```shell
lf -remote "send $id :toggle file1 file2; tag-toggle; unselect"
```

That being said, tags are supposed to just be visual markers for user-defined purposes (e.g. mark videos that have already been watched), and are persisted even after `lf` quits. Is there a reason why you can't use file selections for scripting purposes? In other words, select files using `toggle` and then access them using the `fx` environment variable.

**Comment by jpggithub** [comment]
Thanks for your useful answer.

I was unaware about the use of file selections out of `lf`! The `unselect` command would be perfect.

So I try to obtain a list of the files selected by this way:

`lf -remote 'send $echo -e $fx > /tmp/files-selected.txt'`

The problem is that there is only one line that list all the files selected instead of one line by file. It is not convenient if the name of some files have spaces.

I use `filesep="\n"` and `echo $lf_filesep` give one empty line (as attended).

With this configuration, do you see why I can't obtain one name of file by line?

**Reply by joelim-work** [reply]
You need to quote `$fx`, otherwise word-splitting is performed and passed as individual arguments to `echo` which will then print them all on a single line:

```shell
lf -remote 'send $echo -e "$fx" > /tmp/files-selected.txt'
```

Quoting variables is good practice in general, and forgetting to do so is a common pitfall.

**Comment by jpggithub** [comment]
Thank you very much, it works well like that!
